src/slice.cr

```diff
@@ -566,6 +562,12 @@ struct Slice(T)
   protected def check_writable
     raise "Can't write to read-only Slice" if @read_only
   end
+
+  protected def check_size(count : Int)
+    unless 0 <= count <= size
+      raise IndexError.new
```

Comment:
I'm not sure that `IndexError` is the most appropriate here...

Reply:
Why not? It's raised when trying to access an index in the slice that is not in its range.

Reply:
@Sija  I saw that `IndexError.new` is expected to be raised for Slice size issues (destination of copy, index,...) it's also mentioned in the docs for example:
- [Slice#copy_to](https://crystal-lang.org/api/0.27.2/Slice.html#copy_to%28target%3Aself%29-instance-method)
- [Slice#copy_from](https://crystal-lang.org/api/0.27.2/Slice.html#copy_from%28source%3Aself%29-instance-method)

Reply:
Fair 'nuff, I had my doubts because `Slice#copy_{to,from}` methods, which do the bounds-checking, seem to me like they should throw a dedicated Exception class instead, but maybe it's just me ;)